huaweicloud/services/fgs/resource_huaweicloud_fgs_trigger.go

```diff
@@ -0,0 +1,624 @@
+	event["name"] = d.Get("obs.0.event_notification_name").(string)
+	if prefix, ok := d.GetOk("obs.0.prefix"); ok {
+		event["prefix"] = prefix.(string)
+	}
+	if suffix, ok := d.GetOk("obs.0.suffix"); ok {
+		event["suffix"] = suffix.(string)
+	}
+
+	return event
+}
+
+func buildSmnEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"topic_urn": d.Get("smn.0.topic_urn").(string),
+	}
+}
+
+func buildDisEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"stream_name":        d.Get("dis.0.stream_name").(string),
+		"sharditerator_type": d.Get("dis.0.starting_position").(string),
+		"max_fetch_bytes":    d.Get("dis.0.max_fetch_bytes").(int),
+		"polling_interval":   d.Get("dis.0.pull_period").(int),
+		"polling_unit":       "ms",
+		"is_serial":          strconv.FormatBool(d.Get("dis.0.serial_enable").(bool)),
+	}
+}
+
+func buildKafkaEventData(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"instance_id": d.Get("kafka.0.instance_id").(string),
+		"batch_size":  d.Get("kafka.0.batch_size").(int),
+		"topic_ids":   utils.ExpandToStringListBySet(d.Get("kafka.0.topic_ids").(*schema.Set)),
+	}
+}
+
+func buildFunctionGraphTriggerParameters(d *schema.ResourceData, config *config.Config) (trigger.CreateOpts, error) {
+	triggerType := d.Get("type").(string)
+
+	opts := trigger.CreateOpts{
+		TriggerTypeCode: triggerType,
+		TriggerStatus:   d.Get("status").(string),
+	}
+
+	switch triggerType {
+	case timingTrigger:
+		opts.EventData = buildTimingEventData(d)
+		opts.EventTypeCode = "MessageCreated"
+	case obsTrigger:
+		opts.EventData = buildObsEventData(d)
+		opts.EventTypeCode = "MessageCreated"
+	case smnTrigger:
+		opts.EventData = buildSmnEventData(d)
+		opts.EventTypeCode = "MessageCreated"
+	case disTrigger:
+		opts.EventData = buildDisEventData(d)
+		opts.EventTypeCode = "MessageCreated"
+	case kafkaTrigger:
+		opts.EventData = buildKafkaEventData(d)
+		opts.EventTypeCode = "MessageCreated"
+	default:
+		return opts, fmtp.Errorf("Parameter 'type' is invalid")
+	}
+	return opts, nil
+}
+
+func resourceFunctionGraphTriggerCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	client, err := config.FgsV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud FunctionGraph v2 client: %s", err)
+	}
+
+	opts, err := buildFunctionGraphTriggerParameters(d, config)
+	if err != nil {
+		return fmtp.Errorf("Error building create options of FunctionGraph: %s", err)
+	}
+	logp.Printf("[DEBUG] The create options is: %#v", opts)
+	urn := d.Get("function_urn").(string)
+	resp, err := trigger.Create(client, opts, urn).Extract()
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud FunctionGraph trigger for function (%s): %s", urn, err)
+	}
+	d.SetId(resp.TriggerId)
+
+	if resp.TriggerTypeCode == kafkaTrigger && resp.Status != d.Get("status").(string) {
```

Comment:
suggest to add a default value for `status`.

Reply:
About empty setting of the status for DMS kafka, add some conditions to make trigger enter ACTIVE status successfully.